sdk/storage/Azure.Storage.Blobs/tests/TransactionalHashingTests.cs

```diff
@@ -0,0 +1,653 @@
+        public async Task BlobClientUploadRejectPrecalculatedHash(TransactionalHashAlgorithm algorithm)
+        {
+            await using DisposingContainer test = await GetTestContainerAsync();
+
+            var client = await MakeBlobClient(test.Container, GetOptions());
+
+            TransactionalHashingTestSkeletons.TestPrecalculatedHashNotAccepted(
+                Recording.Random, algorithm,
+                async (stream, hashingOptions) => await client.UploadAsync(stream, new BlobUploadOptions()
+                {
+                    TransactionalHashingOptions = hashingOptions
+                }));
+        }
+        #endregion
+
+        #region BlockBlobClient PartitionedUpload
+        private static async Task BlockBlobParallelUploadAction(
+            BlockBlobClient blob,
+            Stream stream,
+            UploadTransactionalHashingOptions hashingOptions,
+            StorageTransferOptions transferOptions)
+            => await blob.UploadAsync(stream, new BlobUploadOptions
+            {
+                TransactionalHashingOptions = hashingOptions,
+                TransferOptions = transferOptions
+            });
+
+        [TestCase(TransactionalHashAlgorithm.MD5)]
+        //[TestCase(TransactionalHashAlgorithm.StorageCrc64)] TODO #23578
+        public async Task BlockBlobClientUploadSuccessfulHashVerification(TransactionalHashAlgorithm algorithm)
+        {
+            await using DisposingContainer test = await GetTestContainerAsync();
+
+            await TransactionalHashingTestSkeletons.TestParallelUploadSuccessfulHashComputationAsync(
+                Recording.Random, algorithm, () => GetOptions(), test.Container, MakeBlockBlobClient, BlockBlobParallelUploadAction);
+        }
+
+        [TestCase(TransactionalHashAlgorithm.MD5)]
+        //[TestCase(TransactionalHashAlgorithm.StorageCrc64)] TODO #23578
+        public async Task BlockBlobClientUploadRejectPrecalculatedHash(TransactionalHashAlgorithm algorithm)
+        {
+            await using DisposingContainer test = await GetTestContainerAsync();
+
+            var client = await MakeBlockBlobClient(test.Container, GetOptions());
+
+            TransactionalHashingTestSkeletons.TestPrecalculatedHashNotAccepted(
+                Recording.Random, algorithm,
+                async (stream, hashingOptions) => await client.UploadAsync(stream, new BlobUploadOptions()
+                {
+                    TransactionalHashingOptions = hashingOptions
+                }));
+        }
+        #endregion
+
+        #region BlockBlobClient StageBlock
+        private Task<BlockBlobClient> MakeBlockBlobClient(BlobContainerClient container, BlobClientOptions testClientOptions)
+        {
+            container = InstrumentClient(new BlobContainerClient(container.Uri, GetNewSharedKeyCredentials(), testClientOptions));
+            return Task.FromResult(InstrumentClient(container.GetBlockBlobClient(GetNewBlobName())));
```

Comment:
`container.GetBlockBlobClient()` isn't a async method, we could return a `BlockBlobClient` instead of a `Task<BlockBlobClient>` here

Reply:
Block blobs are the only resource that don't need a `CreateAsync()` call to function. This method is the exception that just needs to fit the `Func` signature in tests.